### ISSUE TITLE
[2020 ICML] Learning De-biased Representations with Biased Representations

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#55]
학습을 할 때, 대부분 single source data를 random split하여 train, test set으로 나누어 학습하여 evaluation을 함

Dataset bias로 충분히 high accuracy를 달성할 수 있다면, 모델은 capacity가 충분하더라도 task에 맞는 complexity를 배우지 않고 bias를 학습하게 됨
(learn 'shortcuts')

이로 인해 모델은 주어진 데이터셋에서 좋은 성능을 내지만, bias shift가 발생한 경우에서 generalise 능력이 떨어지게 됨

해당 논문은 bias를 학습하는 모델 g의 representation과 task에 맞는 complexity를 배우는 f의 representation이 statistically independent하게 만드는 loss term을 추가하여 min-max optimisation으로 해결함

논문 중 problem definition, proposed method, experiments만 설명

### 1.1. Cross-bias generalisation

X : input
S : signal of X
B : bias of X
Y : target

Signals S are the cues essential for the recognition of X as Y

Biases B are cues not essential for the recognition but correlated with the target Y

- In-distribution
    - p(S_train, B_train, Y_train) = p(S_test, B_test, Y_test)
- Cross-domain
    - p(S_train, B_train, Y_train) != p(S_test, B_test, Y_test) and p(B_train) != p(B_test)
    - (B_train, Y_train) = [(wilderness, frog), (wilderness, bird)]
    - (B_test, Y_test) = [(indoors, frog), (indoors, bird)]
- Cross-bias
    - p(B_train) and p(Y_train) is dependent
    - p(B_train, Y_train) != p(B_test, Y_test)
    - Assume p(B_train) = p(B_test) to distinguish from the cross-domain scenario
    - (B_train, Y_train) = [(swamp, frog), (sky, bird)]
    - (B_test, Y_test) = [(sky, frog), (swamp, bird)]

해당 논문은 cross-bias scenario에 대해 해결하고자 함

### 1.2. Existing cross-bias generalisation methods and their assumptions

Cross-bias scenario에서 문제가 되는 부분은 p(B_train)과 p(Y_train)가 dependent하기에, recognition을 하는데 있어 bias B가 viable cue가 된다는 점임

모델은 쉬운 cue인 bias B를 학습하여 generalisability가 저하됨

이를 해결할 수 있는 방법들과 한계점은 다음과 같음

1. Algorithm to disentangle bias B and signal S
    - Bias B와 signal S를 disentangle한다는 것 자체가 unrealistic함
2. Data collection or generative algorithm for p(X|B)
    - Collecting unusual inputs can be expensive
    - Building a generative model with pre-defined bias types may suffer from bias misspecification or the lack of realism
3. Ground truth or predictive algorithm for p(B|X) exists
    - Input X가 주어졌을 때 bias B를 말해줄 수 있는 모델이 있다면 dependency를 제거할 수 있음
    - 대부분의 상황에서 ground truth p(B|X)가 존재하지 않으며 bias type를 규정하는 것 자체도 어려움

Texture bias를 예로 들면 다음과 같은 한계점이 존재함
1. Texture B and shape S cannot easily be disentangled
2. Collecting unusual images or building a generative model p(X|B) is expensive
3. Building the predictive model p(B|X) for texture requires enumeration (classifier) or embedding (regression) of all possible textures, which is not feasible

### 1.3. Our scenario: Capturing bias with a set of models

해당 논문은 3번째 방법을 우회해서 해결하고자 함

즉 p(B|X)를 bias B를 배우게끔 특화된 모델 G를 통해 approximation할 수 있다는 것임

Texture bias를 예로 들면 모델 G는 small receptive field만 가지게 제한하여 학습을 하면 texture에 overfit됨

모델 G가 bias만을 capture하길 바라지만 중요한 signal도 capture하기에, regulariser term으로 만들어 F가 G로 capture된 정보를 무시하지 않게 만듬

## 2. ReBias: Removing bias with bias
<img width="326" alt="image" src="https://user-images.githubusercontent.com/49643709/158025070-eda4014a-4bd1-45b4-9259-8953efd897da.png">
<img width="670" alt="image" src="https://user-images.githubusercontent.com/49643709/158025053-1820e976-2096-4685-a5aa-149c925b7364.png">

HSIC (Hilbert-Schmidt Independence Criterion) : 2 random variable 간의 independence를 나타내는 척도
(0일때 variable은 independent)

#### 3.1.1. Dataset
<img width="334" alt="image" src="https://user-images.githubusercontent.com/49643709/158025620-9a6a4366-e359-4f33-810a-15a9026aaa98.png">

각 digit마다 color를 지정해두고 ρ를 이용하여 biased 정도를 만듬
(ρ의 확률로 지정해둔 color를 digit에 입힘, ρ = 1이면 complete bias, ρ = 0.1이면 un-biased)

#### 3.1.2. Model

Biased model g는 (1x1 conv, BN, ReLU) * 4

De-biased model f는 (7x7 conv, BN, ReLU) * 4

#### 3.1.3. Result
<img width="665" alt="image" src="https://user-images.githubusercontent.com/49643709/158025669-32e8068f-94ef-414b-8355-4d42176973ca.png">
<img width="663" alt="image" src="https://user-images.githubusercontent.com/49643709/158026860-76f614e5-978e-4486-9aff-9a912e916cfe.png">

ReBias achieves robust generalisation across all settings by learning to be different from the representations G

#### 3.1.4. Analysis
<img width="678" alt="image" src="https://user-images.githubusercontent.com/49643709/158025888-addbc62a-6bc0-4a4e-91c6-4f67b70faa13.png">

ReBias uniformly improves the off-diagonals, while not sacrificing the diagonals

<img width="338" alt="image" src="https://user-images.githubusercontent.com/49643709/158025939-6819210b-82ca-4987-a201-c048c0539408.png">

Unbiased accuracy increases dramatically (> 60%) as the HSIC between F and G is minimised during training

We observe that there exists a strong correlation between the HSIC values and unbiased accuracies

#### 3.2.1. Dataset
<img width="541" alt="image" src="https://user-images.githubusercontent.com/49643709/158026281-848b76e2-c937-41ad-a4a0-209a7c342440.png">

Original ImageNet을 사용하기에 scale이 너무 커서, 9-Class ImageNet에서 실험 진행
(subset of imageNet containing 9 super-classes)

<img width="667" alt="image" src="https://user-images.githubusercontent.com/49643709/158026374-2bb3e2d2-ea2d-452d-826f-78b6d553b1b7.png">
<img width="593" alt="image" src="https://user-images.githubusercontent.com/49643709/158026540-01369c3a-7899-4a72-a880-85bfd0b37288.png">

Unbiased dataset을 구축하기 위해 texture feature clustering을 통해 이미지에 대한 texture label을 얻음
(ImageNet pre-trained VGG16의 low-layer feature maps의 gram matrix를 이용하여 texture feature를 얻어 k-means clustering)

#### 3.2.2. Model

Biased model g는 BagNet 18 (몇몇 3x3 conv를 1x1 conv로 바꿔 receptive field를 43으로 맞춤)

De-biased model f는 ResNet 18 (receptive field 435)

#### 3.2.3. Result
<img width="334" alt="image" src="https://user-images.githubusercontent.com/49643709/158026583-fa128f54-5238-4efe-9c46-4761e2fba713.png">
<img width="674" alt="image" src="https://user-images.githubusercontent.com/49643709/158026842-c49a1a5a-484f-4026-98e8-ed9ac8322025.png">

Vanilla models are biased and ReBias leads to less biased models

#### 3.2.4. Analysis
<img width="677" alt="image" src="https://user-images.githubusercontent.com/49643709/158026900-907d8b7b-a020-4975-bd44-ce1b56d55121.png">

#### 3.3.1. Dataset
<img width="675" alt="image" src="https://user-images.githubusercontent.com/49643709/158026916-979b7b07-0318-4228-89a8-a7ae229101f3.png">

Kinetic dataset : biased dataset (bias towards static cues, train)

Mimetics dataset : unbiased dataset

#### 3.3.2. Model
<img width="683" alt="image" src="https://user-images.githubusercontent.com/49643709/158026810-894321ad-a4e6-43b4-8082-6ddb737197bc.png">

Biased model g는 2D ResNet 18

De-biased model f는 3D ResNet 18

#### 3.3.3. Result
<img width="670" alt="image" src="https://user-images.githubusercontent.com/49643709/158026880-7959defd-f8e3-43dc-9239-f41a58ab9991.png">
